### PR TITLE
Bugfix/hhios 402 closing the sheet should show the search directly

### PR DESCRIPTION
**Pull request by NaifAlrashed**
### Description
implement a generic way in the sheet to not allow for duplicate sheets. Specifically, POI sheet now does not allow for stacking of POIs

https://github.com/user-attachments/assets/0d752ee8-582a-455d-a218-19248d022f40

### CODE REVIEW

**Review by AlloTheOne — APPROVED** [review]
amazing as usual, and a note in your PR video it seems like there’s no placeholder for the duration text so every time the duration is loading we see the taps then it hides when the duration loads, I think it's better to add a placeholder or "a skeleton"

File: HudHud/Sheet/SheetStore.swift [hunk 1]
```diff
@@ -101,6 +101,17 @@ final class SheetStore {
         self.sheets = []
         self.navigationCommands.send(.popToRoot(rootSheetData: self.emptySheetData))
     }
+
+    func didShow(_ sheetData: SheetData) {
```
Comment (FatooAJ) [inline HudHud/Sheet/SheetStore.swift:105]: Amazing coding here! It took me some time to understand how it works and what isSameCase does.

File: HudHud/Sheet/SheetStore.swift [hunk 2]
```diff
@@ -196,14 +207,16 @@ enum NavigationCommand {
     case show(SheetData)
     case pop(destinationSheetData: SheetData)
     case popToRoot(rootSheetData: SheetData)
+    case removeSheet(index: Int)
```
Comment (FatooAJ) [inline HudHud/Sheet/SheetStore.swift:210]: At first, I thought this would allow me to pop a specific number of sheets, but then I understood it.

**Review by FatooAJ — APPROVED** [review]
I learned a lot as always ⭐️

**Pull request merged**